pyvista/plotting/plotting/__init__.py

```diff
@@ -0,0 +1,27 @@
+"""Deprecated pyvista.plotting.plotting module."""
+import importlib
+import inspect
+import warnings
+
+from pyvista.core.errors import PyVistaDeprecationWarning
+
+
+def __getattr__(name):
+    module = importlib.import_module('pyvista.plotting.plotter')
+    try:
+        value = inspect.getattr_static(module, name)
```

Comment:
The logic might be fine here, again just making sure that nothing falls through the cracks.

On `main` these were the contents of `pyvista.plotting.plotting`:
```
>>> names
['Actor', 'BasePlotter', 'BoundsLike', 'Camera', 'Color', 'CompositePolyDataMapper', 'DataSetMapper', 'Dict', 'FONTS', 'FieldAssociation', 'FixedPointVolumeRayCastMapper', 'GPUVolumeRayCastMapper', 'KILL_DISPLAY', 'MissingDataError', 'OpenGLGPUVolumeRayCastMapper', 'Optional', 'PickingHelper', 'Plotter', 'PointGaussianMapper', 'Property', 'PyVistaDeprecationWarning', 'RenderWindowInteractor', 'RenderWindowUnavailable', 'Renderer', 'Renderers', 'SUPPORTED_FORMATS', 'ScalarBars', 'SmartVolumeMapper', 'Thread', 'USE_SCALAR_BAR_ARGS', 'UnstructuredGridVolumeRayCastMapper', 'Volume', 'VolumeProperty', 'WidgetHelper', '_ALL_PLOTTERS', '__annotations__', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', '_coerce_pointslike_arg', '_common_arg_parser', '_kill_display', '_vtk', '_warn_xserver', 'abstract_class', 'active_scalars_algorithm', 'algorithm_to_mesh_handler', 'assert_empty_kwargs', 'close_all', 'collections', 'contextmanager', 'convert_array', 'ctypes', 'decimation_algorithm', 'deepcopy', 'export_plotter_vtkjs', 'extract_surface_algorithm', 'get_array', 'get_array_association', 'get_cmap_safe', 'image_from_window', 'io', 'is_pyvista_dataset', 'log', 'logging', 'matplotlib', 'normalize', 'np', 'numpy_to_texture', 'opacity_transfer_function', 'os', 'parse_font_family', 'pathlib', 'platform', 'pointset_to_polydata_algorithm', 'prepare_smooth_shading', 'process_opacity', 'pyvista', 'raise_not_matching', 'run_image_filter', 'scooby', 'set_algorithm_input', 'sys', 'textwrap', 'time', 'triangulate_algorithm', 'uses_egl', 'warnings', 'weakref', 'wrap', 'wraps']
```
All of these are defined on the new `pyvista.plotting.plotter`. Yet from these, here are the ones that are not defined on `pyvista.plotting` of this branch:
```
>>> [name for name in names if not hasattr(pyvista.plotting, name)]
['BoundsLike', 'Dict', 'FieldAssociation', 'FixedPointVolumeRayCastMapper', 'GPUVolumeRayCastMapper', 'KILL_DISPLAY', 'MissingDataError', 'OpenGLGPUVolumeRayCastMapper', 'Optional', 'PickingHelper', 'PointGaussianMapper', 'PyVistaDeprecationWarning', 'RenderWindowInteractor', 'RenderWindowUnavailable', 'Renderers', 'SUPPORTED_FORMATS', 'ScalarBars', 'SmartVolumeMapper', 'Thread', 'USE_SCALAR_BAR_ARGS', 'UnstructuredGridVolumeRayCastMapper', 'Volume', 'VolumeProperty', '_ALL_PLOTTERS', '__annotations__', '_coerce_pointslike_arg', '_common_arg_parser', '_kill_display', '_warn_xserver', 'abstract_class', 'active_scalars_algorithm', 'assert_empty_kwargs', 'collections', 'contextmanager', 'convert_array', 'ctypes', 'decimation_algorithm', 'deepcopy', 'extract_surface_algorithm', 'get_array', 'get_array_association', 'io', 'is_pyvista_dataset', 'log', 'logging', 'matplotlib', 'normalize', 'np', 'os', 'pathlib', 'platform', 'pointset_to_polydata_algorithm', 'prepare_smooth_shading', 'process_opacity', 'pyvista', 'raise_not_matching', 'scooby', 'sys', 'textwrap', 'time', 'triangulate_algorithm', 'uses_egl', 'warnings', 'weakref', 'wrap', 'wraps']
```

It seems to me (but I might have misread something) that all these names are found by `inspect.getattr_static()`, yet the recommended `from pyvista.plotting import Renderers` (or any of the other items) will fail.

Is this the case, and if yes, do we/can we want to change anything about this? Either in the logic here, or about the accessibility of names. And I haven't checked but odds are that the deprecated `utilities` __init__.py` is in a similar situation.

Reply:
Thanks for bringing this up. I'll make sure the `pyvista.plotting` namespace works the same

Reply:
Will add validation testing in a follow up PR

Reply:
Yep. The `pyvista.utilities` module is being deprecated. I had changes here that implemented an interface for this import to issue a deprecation warning, but it ruined the history/blame and I will have to implement as a follow on PR